core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -556,11 +512,135 @@ void toSerializable(DistributedStorage storage)
     parallelExecutor.validate(tasks, getId());
   }
 
-  private TableMetadata getTableMetadata(Scan scan) throws ExecutionException {
-    TransactionTableMetadata metadata = tableMetadataManager.getTransactionTableMetadata(scan);
+  /**
+   * Validates the scan results to check if there is no anti-dependency.
+   *
+   * <p>This method scans the latest data and compares it with the scan results. If there is a
+   * discrepancy, it means that the scan results are changed by another transaction. In this case,
+   * an {@link ValidationConflictException} is thrown.
+   *
+   * <p>Since the validation is performed after the prepare-record phase, the scan might include
+   * prepared records if the transaction has performed operations that affect the scan result. In
+   * such cases, those prepared records can be safely ignored.
+   *
+   * <p>Note that this logic is based on the assumption that identical scans return results in the
+   * same order, provided that the underlying data remains unchanged.
+   *
+   * @param storage a distributed storage
+   * @param scan the scan to be validated
+   * @param results the results of the scan
+   * @throws ExecutionException if a storage operation fails
+   * @throws ValidationConflictException if the scan results are changed by another transaction
+   */
+  private void validateScanResults(
+      DistributedStorage storage, Scan scan, LinkedHashMap<Key, TransactionResult> results)
```

Comment:
How about using `original` and `latest` to clearly differentiate the first scan results before the read validation and the second scan results during the read validation?
`original` and `rescanned` can be another option.

Reply:
Fixed in bcfcf068e78243934344f94ba6cbb93e45e00c57. Thanks!